src/txn/txn_rollback_to_stable.c

```diff
@@ -286,13 +261,16 @@ __rollback_row_ondisk_fixup_key(WT_SESSION_IMPL *session, WT_PAGE *page, WT_ROW
     wt_timestamp_t hs_durable_ts, hs_start_ts, hs_stop_durable_ts, newer_hs_durable_ts;
     uint64_t hs_counter, type_full;
     uint32_t hs_btree_id;
+    uint8_t *memp;
     uint8_t type;
     char ts_string[4][WT_TS_INT_STRING_SIZE];
     bool valid_update_found;
 #ifdef HAVE_DIAGNOSTIC
     bool first_record;
 #endif
 
+    if (page == NULL && ref != NULL)
+        page = ref->page;
```

Comment:
Instead of if check, can you change it to an assert?
if (page == NULL) {
    WT_ASSERT(session, ref != NULL)
    page = ref->page;
}

Reply:
Fixed.